views/calendar/calendar.js

```diff
@@ -130,285 +124,62 @@ window.calendarView = class calendarView extends View {
-                        year: date.split("-")[0],
-                        month: date.split("-")[1],
-                        day: date.split("-")[2],
-                        marriage: `<a href="https://www.wikitree.com/wiki/${link}">${spouse}</a>`,
-                    });
-                } else if (getMonth(date) == "Aug" && getMonth(date) != undefined) {
-                    AllMonths.Aug.push({
-                        id: person.Name,
-                        name: person.ShortName,
-                        text: text,
-                        date: date,
-                        year: date.split("-")[0],
-                        month: date.split("-")[1],
-                        day: date.split("-")[2],
-                        marriage: `<a href="https://www.wikitree.com/wiki/${link}">${spouse}</a>`,
-                    });
-                } else if (getMonth(date) == "Sep" && getMonth(date) != undefined) {
-                    AllMonths.Sep.push({
-                        id: person.Name,
-                        name: person.ShortName,
-                        text: text,
-                        date: date,
-                        year: date.split("-")[0],
-                        month: date.split("-")[1],
-                        day: date.split("-")[2],
-                        marriage: `<a href="https://www.wikitree.com/wiki/${link}">${spouse}</a>`,
-                    });
-                } else if (getMonth(date) == "Oct" && getMonth(date) != undefined) {
-                    AllMonths.Oct.push({
-                        id: person.Name,
-                        name: person.ShortName,
-                        text: text,
-                        date: date,
-                        year: date.split("-")[0],
-                        month: date.split("-")[1],
-                        day: date.split("-")[2],
-                        marriage: `<a href="https://www.wikitree.com/wiki/${link}">${spouse}</a>`,
-                    });
-                } else if (getMonth(date) == "Nov" && getMonth(date) != undefined) {
-                    AllMonths.Nov.push({
-                        id: person.Name,
-                        name: person.ShortName,
-                        text: text,
-                        date: date,
-                        year: date.split("-")[0],
-                        month: date.split("-")[1],
-                        day: date.split("-")[2],
-                        marriage: `<a href="https://www.wikitree.com/wiki/${link}">${spouse}</a>`,
-                    });
-                } else if (getMonth(date) == "Dec" && getMonth(date) != undefined) {
-                    AllMonths.Dec.push({
+                for (const monthName of MONTHS) {
+                    const [year, month, day] = date.split("-").map((item) => parseInt(item));
+                    if (month === 0 || monthName != MONTHS[month - 1]) continue;
+                    AllMonths[monthName.slice(0, 3)].push({
```

Comment:
I've just realized that this entire for loop is not necessary...

```suggestion
                    const [year, month, day] = date.split("-").map((item) => parseInt(item));                    
                    if (year == 0 || month === 0 || day === 0) return; // we don't know full date -> we don't want his                    
                    AllMonths[MONTHS[month-1].slice(0, 3)].push({ 
```